examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.kt

```diff
@@ -0,0 +1,256 @@
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(
+    name: String?, min: Short, max: Short, out: AtomicLong?, desc: String?, optional: Boolean
+  ) {
+    val arg = Argument(name!!, min, max, out!!, desc, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add a long Integer command argument
+   * @param name The name that will be displayed in the command help
+   * @param min The minimum value of the argv value
+   * @param max The minimum value of the argv value
+   * @param out A pointer to a AtomicLong where the argv value will be stored
+   * @param desc The description of the argument that will be displayed in the command help
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(
+    name: String?, min: Long, max: Long, out: AtomicLong?, desc: String?, optional: Boolean
+  ) {
+    val arg = Argument(name!!, min, max, out!!, desc, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add an IP address command argument
+   * @param name The name that will be displayed in the command help
+   * @param out A pointer to a IPAddress where the argv value will be stored
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(name: String?, out: IPAddress?, optional: Boolean) {
+    val arg = Argument(name!!, out!!, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Add a String command argument
+   * @param name The name that will be displayed in the command help
+   * @param out A pointer to a StringBuffer where the argv value will be stored
+   * @param desc The description of the argument that will be displayed in the command help
+   * @param optional Indicate if an optional argument
+   * @return The number of arguments currently added to the command
+   */
+  fun addArgument(
+    name: String?, out: StringBuffer?, desc: String?, optional: Boolean
+  ) {
+    val arg = Argument(name!!, out!!, desc, optional)
+    addArgumentToList(arg)
+  }
+
+  /**
+   * @brief Initialize command arguments
+   * @param argc The number of arguments to a command (Does not include command itself)
+   * @param args Supplied command-line arguments as an array of String objects.
+   */
+  fun initArguments(argc: Int, args: Array<String>) {
+    var mandatoryArgsCount = 0
+    var currentIndex = 0
+    for (arg in arguments) {
+      if (!arg.isOptional) {
+        mandatoryArgsCount++
+      }
+    }
+    require(argc >= mandatoryArgsCount) { "initArguments: Wrong arguments number: $argc instead of $mandatoryArgsCount" }
+
+    // Initialize mandatory arguments
+    for (i in 0 until mandatoryArgsCount) {
+      initArgument(currentIndex++, args[i])
+    }
+
+    // Initialize optional arguments
+    // Optional arguments expect a name and a value, so i is increased by 2 on every step.
```

Comment:
What should happen if there is a set of optional arguments and there are multiple arguments with the same name?

Reply:
We check the input parameters against pre-defined argument spec

```
      if (args[i]
          .substring(OPTIONAL_ARGUMENT_PREFIX_LENGTH)
        == arguments[currentIndex].name
```

If you input multiple arguments with the same name, the extra one will failed within the check above.